src/compas_robots/viewer/scene/robotmodelobject.py

```diff
@@ -0,0 +1,161 @@
+from typing import Optional
+
+from compas.colors import Color
+from compas.datastructures import Mesh
+from compas.geometry import Transformation
+from compas_viewer.scene import MeshObject
+from compas_viewer.scene import ViewerSceneObject
+
+from compas_robots import Configuration
+from compas_robots import RobotModel
+from compas_robots.scene import BaseRobotModelObject
+
+
+class RobotModelObject(BaseRobotModelObject, ViewerSceneObject):
+    """Viewer scene object for displaying COMPAS Robot geometry.
+
+    Parameters
+    ----------
+    model : :class:`compas_robots.RobotModel`
+        The robot model.
+    configuration : :class:`compas_robots.Configuration`, optional
+        The initial configuration of the robot. Defaults to the zero configuration.
+    show_visual : bool, optional
+        Toggle the visibility of the visual geometry. Defaults to True.
+    show_collision : bool, optional
+        Toggle the visibility of the collision geometry. Defaults to False.
+    hide_coplanaredges : bool, optional
+        True to hide the coplanar edges. It will override the value in the config file.
+    use_vertexcolors : bool, optional
+        True to use vertex color. It will override the value in the config file.
+    **kwargs : dict, optional
+        Additional keyword arguments.
+        For more info, see :class:`compas_viewer.scene.ViewerSceneObject`.
+
+    See Also
+    --------
+    :class:`compas_robots.scene.BaseRobotModelObject`
+    :class:`compas_viewer.scene.ViewerSceneObject`
+    """
+
+    def __init__(
+        self,
+        model: RobotModel,
+        configuration: Optional[Configuration] = None,
+        show_visual: Optional[bool] = None,
+        show_collision: Optional[bool] = None,
+        hide_coplanaredges: Optional[bool] = None,
+        use_vertexcolors: Optional[bool] = None,
+        **kwargs
+    ):
+        self.use_vertexcolors = use_vertexcolors
+        self.hide_coplanaredges = hide_coplanaredges
+        self._show_visual = show_visual or True
+        self._show_collision = show_collision or False
+        self.configuration: Configuration = configuration or model.zero_configuration()
+        super(RobotModelObject, self).__init__(model=model, **kwargs)
+
+        self.visual_objects: list[MeshObject] = self.draw_visual()
+        self.collision_objects: list[MeshObject] = self.draw_collision()
+
+    @property
+    def show_visual(self):
+        return self._show_visual
+
+    @show_visual.setter
+    def show_visual(self, value: bool):
+        if value == self._show_visual:
+            return
+        self._show_visual = value
+        for i, visual_object in enumerate(self.visual_objects):
+            if value:
+                self.scene.tree.add_object(visual_object, self.visual_objects[i - 1] if i > 0 else self)
```

Comment:
I think the conditional inside the method call is hard to read, please assign the value of that to a variable and then just pass it to `add_object`